lib/controllers/cats.js

```diff
@@ -1,15 +1,17 @@
 const { Router } = require('express');
-const { cats } = require('../../data/cats');
+const Cat = require('../models/Cat');
 const router = Router();
 
 router
-  .get('/:id', (req, res) => {
-    // console.log('id param:', req.params.id);
-    const cat = cats.find((cat) => cat.id === req.params.id);
+  .get('/:id', async (req, res) => {
+    console.log('id param:', req.params.id);
+    const cat = await Cat.getById(req.params.id);
+    console.log(cat);
     res.json(cat);
   })
-  .get('/', (req, res) => {
+  .get('/', async (req, res) => {
     // console.log(cats);
+    const cats = await Cat.getAll();
     const ids = cats.map((cat) => ({ id: cat.id, name: cat.name }));
```

Comment:
You don't need line 15 anymore.  Your get all is handling the mapping inside your Cat model.